core/events/consumers.py

```diff
@@ -0,0 +1,153 @@
+                    continue
+                if message.event.match(task.acts_on):
+                    self.logger.info(f"Running task {task.name}")
+                    task.run(message.event)
+        except Exception:
+            self.logger.exception(
+                f"[PID:{os.getpid()}] - Error processing message in events queue with {body}"
+            )
+        received_message.ack()
+
+
+class LogWorker(Worker):
+    def __init__(self, connection, queues):
+        super().__init__(LogTask, connection, queues)
+
+    def on_message(self, body, received_message):
+        try:
+            message = LogMessage(**json.loads(body))
+            for task in TaskManager.tasks():
+                if task.enabled is False or task.type != TaskType.log:
+                    continue
+                task.run(message.log)
+        except Exception:
+            self.logger.exception(f"Error processing message in logs queue with {body}")
+        received_message.ack()
+
+
+def event_worker():
+    exchange = Exchange("events", type="direct")
+    queues = [Queue("events", exchange, routing_key="events")]
+    broker = f"redis://{yeti_config.get('redis', 'host')}/"
+    with Connection(broker, heartbeat=4) as conn:
+        worker = EventWorker(conn, queues)
+        worker.run()
+
+
+def log_worker():
+    exchange = Exchange("logs", type="direct")
+    queues = [Queue("logs", exchange, routing_key="logs")]
+    broker = f"redis://{yeti_config.get('redis', 'host')}/"
+    with Connection(broker, heartbeat=4) as conn:
+        worker = LogWorker(conn, queues)
+        worker.run()
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(
+        prog="yeti-consumer", description="Consume events and logs from the event bus"
+    )
+    parser.add_argument(
+        "--concurrency", type=int, default=None, help="Number of consumers to start"
+    )
+    parser.add_argument(
+        "type", choices=["events", "logs"], help="Type of consumer to start"
+    )
+    parser.add_argument("--debug", action="store_true", help="Enable debug logging")
+    args = parser.parse_args()
+    logger.setLevel(logging.DEBUG if args.debug else logging.INFO)
+    if args.type == "events":
+        worker = event_worker
+    elif args.type == "logs":
+        worker = log_worker
+    if not args.concurrency:
+        concurrency = multiprocessing.cpu_count()
+    else:
+        concurrency = args.concurrency
+    if concurrency > 1:
+        logger.info(f"Starting {concurrency} {args.type} workers")
+        try:
+            processes = []
+            for i in range(concurrency):
+                name = f"{args.type}-worker-{i}"
+                p = multiprocessing.Process(target=worker, name=name)
+                p.start()
+                logger.info(f"Started {p.name} pid={p.pid}")
+                processes.append(p)
+            for p in processes:
+                p.join()
+        except KeyboardInterrupt:
```

Comment:
Same here, I think we only need to wrap the loop + join() call?

Reply:
For this one, totally makes sense

Reply:
fixed in 6656b1e7da0e0cc53a423052fdcaa1844175487b